smack-integration-test/src/main/java/org/jivesoftware/smackx/caps/EntityCapsTest.java

```diff
@@ -91,6 +92,7 @@ public void tearDown() throws NotConnectedException, InterruptedException {
     }
 
     @SmackIntegrationTest
+    @SpecificationReference(document = "XEP-0115")
```

Comment:
I am using this code line as example.

I would split this annotation into two: `SpecificationDocumentReference` and `SpecificationReference`. The former `SpecificationDocumentReference` should be used on the class level, while the `SpecificationReference` should be used on the method level. `SpecificationDocumentReference` only takes a required document attribute.

The idea is that you only need to add `SpecificationDocumentReference` to sinttest classes. This has the advantage of reducing boilerplate *and* being less prone to forgetting correctly tagging tests newly added to a class.

I first though that we could remove the `document` property from `SpecificationReference` then. Not sure if there are reasonable cases where you have a method belonging to a sinttest class of XEP-A which is actually testing XEP-B. However, I could image that this case will come up, especially if you have cross cutting concerns between XEPs. We could cross that bridge when we come to it.

Reply:
I prefer to keep the reference to the document and the reference to the section within the document in one annotation. That makes the annotation self-sufficient: that's easier to parse, for both humans as well as in code.

It is also _less_ error prone: when keeping document & section together, any error most likely introduces a _missing_ reference, but when splitting them up, it is a lot easier to create an _incorrect_ reference (by mis-matching document and section). That's generally a worse.

Reply:
> I prefer to keep the reference to the document and the reference to the section within the document in one annotation.

I prefer splitting this to avoid redundant information.

I also noticed that we could probably, and should if possible, re-use the `@SmackIntegrationTest` annotation. So `@SpecificationReference` could become an annotation only containing `document` which must only be used on classes, not methods.

Reply:
Both changes have now been applied. I want my dislike for this approach on the record though :)